### PR TITLE
Added Irish locale formats.

### PR DESCRIPTION
**Pull request by mjfwalsh**

### CODE REVIEW

**Comment by felixxm** [comment]
@mjfwalsh Thanks for this patch :+1: , however adding a new language requires Transifex setup first. Do you intend to start a Irish English translation for Django and take the role of Irish English language coordinator? If you do, we can set it up on Transifex. Please give us your Transifex username.

**Comment by mjfwalsh** [comment]
I don't think translations are really necessary.

I gather from this page

https://docs.djangoproject.com/en/dev/internals/contributing/localizing/#formats

that I don't have to start a Transifex account just to do date and time formats.

I  note that there are already three localisations without translations:

Porto-Rican Spanish ([es_PR](https://github.com/django/django/tree/main/django/conf/locale/es_PR)), Nicaraguan Spanish ([es_NI](https://github.com/django/django/tree/main/django/conf/locale/es_NI)), and Swiss German ([de_CH](https://github.com/django/django/tree/main/django/conf/locale/de_CH)).

I can start a ticket if that's your preferred way of doing things.

**Comment by felixxm** [comment]
> I don't think translations are really necessary.
> 
> I gather from this page
> 
> https://docs.djangoproject.com/en/dev/internals/contributing/localizing/#formats
> 
> that I don't have to start a Transifex account just to do date and time formats.
> 
> I note that there are already three localisations without translations:
> 
> Porto-Rican Spanish ([es_PR](https://github.com/django/django/tree/main/django/conf/locale/es_PR)), Nicaraguan Spanish ([es_NI](https://github.com/django/django/tree/main/django/conf/locale/es_NI)), and Swiss German ([de_CH](https://github.com/django/django/tree/main/django/conf/locale/de_CH)).
> 
> I can start a ticket if that's your preferred way of doing things.

This is something new to me :thinking: , I've never merged formats without translations (\cc @claudep).

**Comment by claudep** [comment]
Sure, when only the formats are specific for some country, it's fine to only commit `formats.py`.

File: django/conf/locale/en_IE/formats.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# see https://docs.python.org/library/datetime.html#strftime-strptime-behavior
+DATE_INPUT_FORMATS = [
+    "%d/%m/%Y",  # '25/10/2006'
+    "%d/%m/%y",  # '25/10/06'
+    # "%b %d %Y",  # 'Oct 25 2006'
+    # "%b %d, %Y",  # 'Oct 25, 2006'
+    "%d %b %Y",  # '25 Oct 2006'
+    "%d %b, %Y",  # '25 Oct, 2006'
+    # "%B %d %Y",  # 'October 25 2006'
+    # "%B %d, %Y",  # 'October 25, 2006'
```
Comment (felixxm) [inline django/conf/locale/en_IE/formats.py:24]:
```suggestion
    "%d %b %Y",  # '25 Oct 2006'
    "%d %b, %Y",  # '25 Oct, 2006'
```

File: django/conf/locale/en_IE/formats.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# The *_FORMAT strings use the Django date format syntax,
+# see https://docs.djangoproject.com/en/dev/ref/templates/builtins/#date
+DATE_FORMAT = "j M Y"  # '25 Oct 2006'
+TIME_FORMAT = "H:i"  # '14:30'
+DATETIME_FORMAT = "j M Y, H:i"  # '25 Oct 2006, 14:30'
+YEAR_MONTH_FORMAT = "F Y"  # 'October 2006'
+MONTH_DAY_FORMAT = "j F"  # '25 October'
+SHORT_DATE_FORMAT = "d/m/Y"  # '25/10/2006'
+SHORT_DATETIME_FORMAT = "d/m/Y H:i"  # '25/10/2006 14:30'
+FIRST_DAY_OF_WEEK = 0  # Sunday
```
Comment (felixxm) [inline django/conf/locale/en_IE/formats.py:12]: Are you sure it's not Monday?
- https://savvytime.com/week-number/ireland/2023
- https://www.timeanddate.com/calendar/?year=2023&country=32

Comment (mjfwalsh) [reply django/conf/locale/en_IE/formats.py:12]: You do admittedly see calendars beginning on Monday, but in my experience calendars beginning with Sunday are by far the most common.

* https://www.irishcalendars.ie/collections/large-format-irish-calendars/products/large-real-ireland-calendar-2022

**Comment by tom-moran** [comment]
@mjfwalsh Nice one adding this, the only setting I would wonder about here is `FIRST_DAY_OF_WEEK` - I would have thought it was Monday, but maybe that is just in my own head! I see conflicting results from a quick search online, not sure if there is an official take on this!

**Comment by felixxm** [comment]
> @mjfwalsh Nice one adding this, the only setting I would wonder about here is `FIRST_DAY_OF_WEEK` - I would have thought it was Monday, but maybe that is just in my own head! I see conflicting results from a quick search online, not sure if there is an official take on this!

@tom-moran Thanks for checking :+1: I also had doubts about `FIRST_DAY_OF_WEEK` see https://github.com/django/django/pull/16641#discussion_r1133653147.

**Comment by tom-moran** [comment]
> > @mjfwalsh Nice one adding this, the only setting I would wonder about here is `FIRST_DAY_OF_WEEK` - I would have thought it was Monday, but maybe that is just in my own head! I see conflicting results from a quick search online, not sure if there is an official take on this!
> 
> @tom-moran Thanks for checking 👍 I also had doubts about `FIRST_DAY_OF_WEEK` see [#16641 (comment)](https://github.com/django/django/pull/16641#discussion_r1133653147).

Hi @felixxm @mjfwalsh I didn't want to approve this until the above was decided, I did some more digging and couldn't find anything further specific to Ireland. If it helps, from https://en.wikipedia.org/wiki/Gregorian_calendar, which Ireland seems to follow:
> "...Opinions vary about the numbering of the days of the week. ISO 8601, in common use worldwide, starts with Monday=1..."
Otherwise happy to +1

**Comment by felixxm** [comment]
So.. do we reach a consensus to use Monday? :)

**Review by tom-moran — APPROVED** [review]

**Pull request merged**